bfsscp/bfsscp.go

```diff
@@ -0,0 +1,376 @@
+// Package bfsscp abstracts an SSH/SCP workflow
+//
+// When imported, it registers both `scp://` and `ssh://` global scheme resolver and can be used like:
+//
+//   import (
+//     "github.com/bsm/bfs"
+//
+//     _ "github.com/bsm/bfs/bfsscp"
+//   )
+//
+//   func main() {
+//     ctx := context.Background()
+//     b, _ := bfs.Connect(ctx, "ssh://user:pass@hostname:22/path/to/root?tmpdir=%2Fcustom%2Ftmp")
+//     f, _ := b.Open(ctx, "file/within/root.txt")
+//     ...
+//   }
+//
+// bfs.Connect supports the following query parameters:
+//
+//   tmpdir - custom temp dir
+//
+package bfsscp
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/bmatcuk/doublestar"
+	"github.com/bsm/bfs"
+	"github.com/bsm/bfs/internal"
+	"github.com/kr/fs"
+	"github.com/pkg/sftp"
+	"go.uber.org/multierr"
+	"golang.org/x/crypto/ssh"
+)
+
+// Register allows for registering more schemes with SCP
+func Register(_ context.Context, u *url.URL) (bfs.Bucket, error) {
```

Comment:
you should make this private

Reply:
It's public to allow for more schemes to be added externally.

Reply:
nah, we shouldn't do this dynamically, I don't know why someone would want a custom scheme?